### ISSUE TITLE
ArgumentNullException reading Patient

### ISSUE THREAD

**Issue by airiarte** [FirelyTeam/fhir-net-api#871]
Hi, I'm getting **System.ArgumentNullException in parameter value** trying to read a patient with Client.Read<Patient>("Patient/10"). It seems this error happens when a Patient has contacts. The patient JSON (obtained with SOAPUI) is as follows: 

{
   "resourceType": "Patient",
   "id": {"value": "10"},
   "identifier": [   {
      "type": {"coding": [      {
         "code": {"value": "2"},
         "display": {"value": "D.N.I"}
      }]},
      "value": {"value": "20202020"}
   }],
   "name": [   {
      "family": {"value": "PEREZ"},
      "given": [{"value": "JUAN MANUEL"}]
   }],
   "gender": {"value": "male"},
   "birthDate": {"value": "1968-09-09T04:00:00Z"},
   "deceasedBoolean": {"value": false},
   "address": [   {
      "line": [{"value": " ANTINACO "}],
      "postalCode": {"value": "5361"}
   }],
   "contact": [   {
      "id": {"value": "6"},
      "relationship": [{"coding": [      {
         "code": {"value": "8"},
         "display": {"value": "sin definir"}
      }]}],
      "name":       {
         "family": {"value": "PEREZ"},
         "given": [{"value": "MARTA"}]
      }
   }]
}

Thank you

**Comment by mbaltus** [comment]
Hi! Looking at it, it seems that the resource data is incorrect. There should not be an id on the contact part, which could be the cause of the exception.
Could you email me at fhirapi@fire.ly with more details, if this is not the cause? Please tell me the version of the API you are using, and the details of the exception.

**Comment by wmrutten** [comment]
Still, even if the input data is invalid, I would expect the caller to receive some concrete serialization format exception, not an `ArgumentNullException` triggered by some internal logic in the API.

**Comment by ewoutkramer** [comment]
I am pretty sure this is an old version of the API, I hope the reporter will get back with the version he used.

**Comment by mbaltus** [comment]
The cause was the server not responding with valid FHIR json. @wmrutten: I agree that a more comprehensive error message might help, but there's a limit to what you can do with invalid data. You could check with the data given, to see if you can improve the message.

**Comment by ewoutkramer** [comment]
I think we should improve the message indeed. It's so broken I'd call this a bug.

**Comment by marcovisserFurore** [comment]
In version 1.2.0-beta2 this message was already much clearer:

> Type checking the data: Encountered unknown element 'value' while parsing

I've change this message by proving also a location of the parsing error:

```
Type checking the data: Encountered unknown element 'value' at location 'Patient.id[0].value[0]' while parsing

```

**Comment by ewoutkramer** [comment]
Fixed by #908
